### PR TITLE
Add method to open a file from scripts

### PR DESCRIPTION
**Pull request by creativitRy**
Hi, CaptainChaos. I implemented a method to easily load files from scripts. I don't know how to install dependencies :(, so this is untested, but theoretically it should work.
From a script, I would use
`var fileDir = org.pepsoft.worldpainter.tools.scripts.ScriptRunner.getFilePath("Open heightmap", "Images", "png jpg jpeg");`

### CODE REVIEW

**Comment by frogocomics** [comment]
Your coding conventions don't fit in with the rest of WorldPainter. Use egyptian brackets

File: WorldPainter/WPGUI/src/main/java/org/pepsoft/worldpainter/tools/scripts/ScriptRunner.java [hunk 1]
```diff
@@ -379,6 +382,25 @@ public void write(@NotNull char[] cbuf, int off, int len) throws IOException {
...
+     * @param extensionName name of list of allowed extensions
+     * @param extensions extension names without the dot in front separated with a space
+     * @return absolute path to the selected file or a blank String if canceled
+     */
+	public static String getFilePath(String title, String extensionName, String extensions) {
+		FileFilter filter = new FileNameExtensionFilter(extensionName, extensions.split(" "));
+		
+		File file = FileUtils.selectFileForOpen(fileWindow, title, null, filter);
+
+		if (file != null)
```
Comment (frogocomics) [inline WorldPainter/WPGUI/src/main/java/org/pepsoft/worldpainter/tools/scripts/ScriptRunner.java]: Add brackets

**Comment by Captain-Chaos** [comment]
@creativitRy, @frogocomics, thanks for the pull requests. I appreciate your enthusiasm, but I should point out that I'm currently not (yet) accepting pull requests. That may change and I will certainly take your requests and suggested implementations seriously, I just want to set the right expectations.

**Comment by Captain-Chaos** [comment]
I don't like to expose internal APIs to scripts, however I appreciate that a convenient way to select files would be handy. I actually already started an attempt to provide a more convenient way of providing script parameters; take a look at the ScriptDescriptor and ParameterDescriptor stuff in ScriptRunner. It already somewhat works; if you include a comment block at the top of your script (before anything else) looking like this, WorldPainter will pick it up and create fields for the parameters on the script execution dialog:

```
// script.name=Script Name
// script.description=Script description\nwith line separators
// script.param.intParam.type=INTEGER
// script.param.intParam.default=-100
// script.param.anotherVar.type=STRING
// script.param.anotherVar.default=Test
```

The `intParam` and `anotherVar` parts are the parameter names, which can be arbitrary identifiers. The parameters will be available to the script in the `params` associative array. There is also a FILE type, but I have yet to implement the actual file selection.

You've triggered me to finish this stuff and I will commit the work in the next few days. When I do, take a look and see if it might be good enough for your purposes.

**Comment by creativitRy** [comment]
Thank you! Something like this would be really helpful.

**Comment by Captain-Chaos** [comment]
I checked in the finished script descriptor stuff. Here is a demo script which you can run from inside WorldPainter (the wpscript command doesn't support script descriptors yet, but I will add it to that as well):

```javascript
// -- This is the script descriptor. It must be the at the start of the script,
// -- *before* any non-empty, non-comment lines. Any lines which don't look like
// -- a key-value pair will be ignored, but you can start script descriptor
// -- comments with a -- just to make sure they won't be mistaken for a
// -- key-value pair.
//
// -- The name of the script (optional):
// script.name=Test
//
// -- A short description which will be shown on the screen (optional):
// script.description=Description of test\nwith optional line endings
//
// -- Script parameter descriptors in the form:
// -- script.param.<paramName>.<property>=<value>:
// script.param.firstParam.type=integer
// script.param.firstParam.description=Description of firstParam. Will be shown as tooltip
//
// script.param.param2.type=string
// script.param.param2.description=Parameter names are arbitrary identifiers
//
// script.param.anotherParam.type=percentage
// script.param.anotherParam.description=Parameters can have default values
// script.param.anotherParam.default=50
//
// script.param.file1.type=file
// script.param.file1.description=Parameters can be optional
// script.param.file1.optional=true
//
// script.param.demoParam.type=boolean
// script.param.demoParam.description=Parameters can have different display names
// script.param.demoParam.displayName=Demo parameter
// script.param.demoParam.default=true
//
// script.param.paramFloat.type=float
// script.param.paramFloat.default=12.34
//
// -- The following line is optional; when present the normal command line
// -- parameter text area will be suppressed:
// script.hideCmdLineParams=true

var firstParam = params['firstParam'];
var param2 = params['param2'];
var anotherParam = params['anotherParam'];
var file1 = params['file1'];
var demoParam = params['demoParam'];
var paramFloat = params['paramFloat'];

print('firstParam (integer): ' + firstParam);
print('param2 (string): ' + param2);
print('anotherParam (percentage): ' + anotherParam);
print('file1 (file): ' + ((file1 == null) ? '<not selected>' : file1));
print('demoParam (boolean): ' + demoParam);
print('paramFloat (float): ' + paramFloat);
```

Let me know if this is good enough for your file selection purposes!

**Comment by creativitRy** [comment]
This is perfect! Thank you.

**Comment by Captain-Chaos** [comment]
Glad to hear it! In that case I'm going to close this pull request.

**Pull request closed without merging**